sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/PodInstanceRequirement.java

```diff
@@ -88,11 +127,20 @@ public PodInstance getPodInstance() {
         return environment == null ? Collections.emptyMap() : environment;
     }
 
-    /**
-     * Returns whether this pod is replacing a previous instance that failed permanently. In this case new resources
-     * should be created, even though (now failed) TaskInfos are still present in the StateStore.
-     */
-    public boolean isPermanentReplacement() {
-        return isPermanentReplacement;
+    public RecoveryType getRecoveryType() {
+        return recoveryType;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Seconded. Pretty uncommon, except for things that can be serialized to and from strings.